diff --git a/src/rogue/hardware/axi/AxiStreamDma.cpp b/src/rogue/hardware/axi/AxiStreamDma.cpp
index 8c98ffaa..9055be52 100755
--- a/src/rogue/hardware/axi/AxiStreamDma.cpp
+++ b/src/rogue/hardware/axi/AxiStreamDma.cpp
@@ -318,6 +318,10 @@ void rha::AxiStreamDma::runThread() {
 
    log_->logThreadId();
 
+   // Workaround to race condition bug
+   // reported in ESROGUE-353
+   sleep(1);
+
    try {
       while(1) {
 
